pkgs/tools/filesystems/rar2fs/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv
+, fetchurl
+, fuse
+}:
+
+with stdenv.lib;
+
+let
+
+  version = "1.28.0";
+
+  unrar-src = fetchurl {
+    url = "https://www.rarlab.com/rar/unrarsrc-5.8.5.tar.gz";
+    sha256 = "0abvz6vv8kr416fphysfbwgxc6hyf1bpnd0aczfv7j3vc8x949d7";
+  };
+
+in
+
+stdenv.mkDerivation {
+  pname = "rar2fs";
+  inherit version;
+  inherit unrar-src;
+
+  prePhases = [ "prePhase" ];
+  prePhase = ''
+    tar xzvf ${unrar-src}
+    cd unrar
+    make lib
+  '';
+
+  buildInputs = [ fuse ];
+
+  configureFlags = [
+    "--with-unrar=/build/unrar/"
+    "--with-unrar-lib=/build/unrar/"
+  ];
+
+  src = fetchurl {
+    url = "https://github.com/hasse69/rar2fs/releases/download/v${version}/rar2fs-${version}.tar.gz";
+    sha256 = "02m8ihz34qnrv57lrif1qr8ch21fg66cyjmk40gafn8zngnqfwlj";
+  };
+
+  meta = {
+    description = "FUSE file system for reading RAR archives";
+    homepage = https://hasse69.github.io/rar2fs/;
+    license = licenses.unfree;
```

Comment:
You can use `gpl3Plus` here if you depend on `unrar` ;)

Reply:
Since the build script is dependant on **unrarsrc**-5.8.5.tar.gz from https://www.rarlab.com/rar/, that makes sense :)

Reply:
Yes, this was assuming that we can re-use `unrar` from Nixpkgs which already has the correct license (and if that wouldn't be possible this could also be a list like `with licenses; [ gpl3Plus unfree]` - but in this case we should just depend on `unrar`).